addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,379 @@
+
+    def send_alldevices(self): # Send device to Odoo
+        server = get_odoo_server_url()
+        if server:
+            iot_box = {'name': socket.gethostname(),'identifier': get_mac_address(), 'ip': get_ip(), 'token': get_token()}
+            devices_list = {}
+            for device in iot_devices:
+                identifier = iot_devices[device].device_identifier
+                devices_list[identifier] = {
+                    'name': iot_devices[device].device_name,
+                    'type': iot_devices[device].device_type,
+                    'connection': iot_devices[device].device_connection,
+                }
+            data = {
+                'params': {
+                    'iot_box' : iot_box,
+                    'devices' : devices_list,
+                }
+            }
+            urllib3.disable_warnings()
+            http = urllib3.PoolManager(cert_reqs='CERT_NONE')
+            try:
+                req = http.request(
+                    'POST',
+                    server + "/iot/setup",
+                    body = json.dumps(data).encode('utf8'),
+                    headers = {'Content-type': 'application/json', 'Accept': 'text/plain'}
+                )
+            except:
+                _logger.warning('Could not reach configured server')
+        else:
+            _logger.warning('Odoo server not set')
+
+    def usb_loop(self):
+        usb_devices = {}
+        devs = core.find(find_all=True)
+        for dev in devs:
+            path =  "usb_%04x:%04x_%03d_%03d_" % (dev.idVendor, dev.idProduct, dev.bus, dev.address)
+            iot_device = IoTDevice(dev, 'usb')
+            usb_devices[path] = iot_device
+        return usb_devices
+
+    def video_loop(self):
+        camera_devices = {}
+        videos = glob('/dev/video*')
+        for video in videos:
+            vd = open(video, 'w')
+            cp = v4l2.v4l2_capability()
+            ioctl(vd, v4l2.VIDIOC_QUERYCAP, cp)
+            cp.interface = video
+            iot_device = IoTDevice(cp, 'video')
+            camera_devices[cp.bus_info.decode('utf-8')] = iot_device
+        return camera_devices
+
+    def printer_loop(self):
+        printer_devices = {}
+        devices = conn.getDevices()
+        for path in [printer_lo for printer_lo in devices if devices[printer_lo]['device-make-and-model'] != 'Unknown']:
+            if 'uuid=' in path:
+                serial = sub('[^a-zA-Z0-9 ]+', '', path.split('uuid=')[1])
+            elif 'serial=' in path:
+                serial = sub('[^a-zA-Z0-9 ]+', '', path.split('serial=')[1])
+            else:
+                serial = sub('[^a-zA-Z0-9 ]+', '', path)
+            devices[path]['identifier'] = serial
+            devices[path]['url'] = path
+            iot_device = IoTDevice(devices[path], 'printer')
+            printer_devices[serial] = iot_device
+        return printer_devices
 
-class USBDeviceManager(Thread):
-    devices = {}
     def run(self):
```

Comment:
need a docstring
thread that will check connected/disconnected device, load drivers if needed and contact the odoo server with the updates